govcd/lbvirtualserver_test.go

```diff
@@ -73,6 +73,16 @@ func (vcd *TestVCD) Test_LBVirtualServer(check *C) {
 	check.Assert(createdLbVirtualServer.ApplicationRuleIds, DeepEquals, lbVirtualServerConfig.ApplicationRuleIds)
 	check.Assert(createdLbVirtualServer.DefaultPoolId, Equals, lbVirtualServerConfig.DefaultPoolId)
 
+	// Try to delete child components and expect a a well parsed NSX error
+	err = edge.DeleteLbServiceMonitorById(serviceMonitorId)
+	check.Assert(err, ErrorMatches, ".*Fail to delete objectId .* for it is used by .*")
```

Comment:
Make the regexp more aggressive by checking that there is actually an ID and an owner, as suggested above

Reply:
Thanks. Improved and tested at least on 9.5 and 9.7